File: app/src/main/java/net/tjado/usbgadget/DeviceInfoMapComparator.java
package net.tjado.usbgadget;

import java.util.Comparator;
import java.util.TreeMap;

public class DeviceInfoMapComparator implements Comparator<String> {

    @Override
    public int compare(String s1, String s2) {
        if (s1.equalsIgnoreCase("KERNEL_VERSION")) {
            return -1;
        } else if (s2.equalsIgnoreCase("KERNEL_VERSION")) {
            return 1;
        } else {
            return s1.compareTo(s2);
        }
    }
}

File: app/src/main/java/net/tjado/usbgadget/DeviceInfoMapComparator.kt
package net.tjado.usbgadget

import java.util.Comparator

class DeviceInfoMapComparator : Comparator<String> {
    override fun compare(s1: String, s2: String): Int {
        return if (s1.equals("KERNEL_VERSION", ignoreCase = true)) {
            -1
        } else if (s2.equals("KERNEL_VERSION", ignoreCase = true)) {
            1
        } else {
            s1.compareTo(s2)
        }
    }
}